### PR TITLE
"hack-on" tweaks

### PR DESCRIPTION
**Pull request by tomsmalley**
Merged develop into the ghcide-compat branch
Renamed hack -> interpret
Use set-like merge for PathTree
Add slightly cleaner debugging for PathTree
Canonicalize excluded obelisk directories to match PathTree

cc @3noch 

I have:

  - [ ] Based work on latest `develop` branch
  - [x] Looked for lint in my changes with `hlint .` (lint found code you did not write can be left alone)
  - [ ] Run the test suite: `$(nix-build -A selftest --no-out-link)`
  - [ ] (Optional) Run CI tests locally: `nix-build release.nix -A build.x86_64-linux --no-out-link` (or `x86_64-darwin` on macOS)

### CODE REVIEW

File: lib/command/src/Obelisk/Command.hs [hunk 1]
```diff
@@ -269,18 +269,21 @@ shellFlags =
...
+            \ override settings for any identical directories given earlier."
           )
-    <|> (, HackOn_NoHackOn) <$>
-          strOption (common <> long "no-hack-on" <> help
+    <|> (, Interpret_NoInterpret) <$>
+          strOption (common <> long "no-interpret" <> help
             "Make packages found in DIR available in the package database (but only when they are used dependencies). \
-            \See help for --hack-on for how the two options are related."
+            \ This will build the packages in DIR before loading GHCi. \
+            \See help for ---on for how the two options are related."
```
Comment (3noch) [inline lib/command/src/Obelisk/Command.hs:286]: Search/replace mishap.

File: lib/command/src/Obelisk/Command/Run.hs [hunk 2]
```diff
@@ -157,51 +173,52 @@ run root hackPaths = do
...
 --
 -- These are a combination of the obelisk predefined local packages,
 -- and any packages that the user has set with the @packages@ argument
 -- to the Nix @project@ function.
-getLocalPkgs :: forall m. MonadObelisk m => FilePath -> PathTree HackOn -> m (Set FilePath)
-getLocalPkgs root hackPaths = do
-  putLog Debug [i|Finding packages with root ${root} and hacking paths ${hackPaths}|]
+getLocalPkgs :: forall m. MonadObelisk m => FilePath -> PathTree Interpret -> m (Set FilePath)
+getLocalPkgs root interpretPaths = do
+  putLog Debug $ [i|Finding packages with root ${root} and interpret paths:|] <> "\n" <> drawPathTree textInterpret interpretPaths
```
Comment (3noch) [inline lib/command/src/Obelisk/Command/Run.hs:209]: You can just inline the expr into the QQ

Comment (3noch) [reply lib/command/src/Obelisk/Command/Run.hs:209]: I think?

Comment (tomsmalley) [reply lib/command/src/Obelisk/Command/Run.hs:209]: I did, but the newline didn't work.

**Pull request merged**